src/main/kotlin/io/csbroker/apiserver/repository/common/RedisRepository.kt

```diff
@@ -39,64 +39,43 @@ class RedisRepository(
         redisTemplate.delete(code)
     }
 
+    @Suppress("UNCHECKED_CAST")
     fun setRank(scoreMap: Map<String, Double>) {
-        redisTemplate.opsForZSet().add(
-            RANKING,
-            scoreMap.map { TypedTuple.of(it.key, it.value) }.toSet(),
+        redisTemplate.execute(
+            object : SessionCallback<Unit> {
+                override fun <K : Any?, V : Any?> execute(operations: RedisOperations<K, V>) {
+                    val stringOperations = operations as RedisOperations<String, String>
+                    stringOperations.multi()
+                    stringOperations.delete(RANKING)
+                    stringOperations.opsForZSet().add(
+                        RANKING,
+                        scoreMap.map { TypedTuple.of(it.key, it.value) }.toSet(),
+                    )
+                    stringOperations.exec()
+                }
+            },
         )
     }
 
     fun getRank(key: String): RankResultDto {
-        var rank: Long? = null
-
         val score = redisTemplate.opsForZSet().score(RANKING, key) ?: 0.0
-        val rankKey = redisTemplate.opsForZSet().reverseRangeByScore(RANKING, score, score, 0, 1)?.first()
+        val rankKey = redisTemplate.opsForZSet().reverseRangeByScore(RANKING, score, score, 0, 1)?.firstOrNull()
 
-        if (rankKey != null) {
-            rank = redisTemplate.opsForZSet().reverseRank(RANKING, rankKey)?.plus(1)
+        val rank = rankKey?.let {
+            redisTemplate.opsForZSet().reverseRank(RANKING, rankKey)?.plus(1)
         }
 
```

Comment:
```suggestion
        val rankKey = redisTemplate.opsForZSet().reverseRangeByScore(RANKING, score, score, 0, 1)?.first()
        val rank = redisTemplate.opsForZSet().reverseRank(RANKING, rankKey!!)?.plus(1)
```
위에 score가 받아져왔다는 건 무조건 하나 이상의 점수가 레디스에 있다는 거니까 !!를 통해 무조건 존재한다는 걸 코드에 내포하는 건 어떨까요?
위와 같이 정리하면 코드가 더 깔끔해질 것 같아요!

Reply:
@ekzm8523 이건 사람마다 의견이 다를 수 있을 것 같아요. 저는 후자인듯?

rank의 score과 key가 확실하게 존재한다는 것을 의미하게 코드 짜기 vs `?` 을 사용하여 null 안정성을 확보하면서 코드짜기.

어떻게 생각하시나요

Reply:
마지막에 변경된 rankKey?.let도 적절해보입니다~ 이렇게 가시져